src/com/jdngray77/htmldesigner/backend/BackgroundTask.kt

```diff
@@ -1,25 +1,52 @@
 package com.jdngray77.htmldesigner.backend
 
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
 /**
  * Runs tasks in the background
  */
-object BackgroundTask {
+object BackgroundTask : Subscriber {
```

Comment:
See about adding some public methods for :
 - Fetching queued tasks
 - Executing tasks

I don't know if it's possible to get this info from the thread pool, but look into it.
They can be used to create a GUI displaying background tasks.